### ISSUE TITLE
tolino vision 5 with cops on Synology: book download problem

### ISSUE THREAD

**Issue by cslondon** [seblucas/cops#517]
Hello,
I have cops version 1.2 running on a Synology NAS DSM7.1, installed via ssh, with apache2.4 and php 7.4. Accessing and downloading books from 2 Kindle paperwhites is fine, when downloading from above tolino I get an error stating the file is defective. I have url-rewriting set to 1 / on  in the local config. 
It appears a known problem with some tolinos, possibly similar to Kobo? 
my specific questions if not a known answer is available:
- how do I enable " AllowOverride All" on Synology Apache 2.4? Apaprenly the .htaccess is not enough or possibly the problem with tolino?
- didn't find any errorlogs - can I activate debugging, where would the logs go?

Thanks.
Carsten

**Comment by cslondon** [comment]
A couple of updates:
- I sideloaded a package some time ago on the Tolino, this allows me to install additional apk packages, including KOREADER, AIREADER, total commander  (which are all  part of the original sideload) but also  wifi plugin for total commander, an alternative webbrowser (called via webbrowser). But not all apk's are install-able, some seems to have problems with requirements.  (BTW: The starting point  of this sideload is https://forum.alles-ebook.de/thread/7130-epos2-vision5-page2-shine3-und-weitere-android-anwendungen/)
- The alternative "via browser"  fails with same errormessage as the internal tolino original browser, but under "download menue" a list  gives the  errormessage which occured which indicates the general:

```
javax.net.ssl Handshake exception :+1: java.security.cert.CertPathValidatorException: Trust anchor for certification path not found.
```
Strange is that this happens when trying to acess within the LAN vua http, not https.

I have no certificates installed on the Synology.
In the future thus type of problem might spread with more and more secure browsers in readers?!

- there possibly is  a webbrowser around which is less security conscious, two other ones didn't install correct via sideload on tolino
- alternative cops access works via KOREADE and its internal OPDS, other reader possibly as well.
 - some alternative reader as well as webbrowser allow for a proxy config (not the tolino internal one), this could be an advantage when http webaccess to cops works, to connect via a vpn mobile phone hotspot - Android doesn't forward VPN connections via hotspot, you need a proxy or similar workaround there. I don't ahre cops by now via internet, only via vpn to outside.
At the moment I use koreader with opds access, but miss the book descriptions and details.  total commander with wifi plugin could be another workaround although via multipls hops ....

**Comment by marioscube** [comment]
@cslondon 

This is not a known problem with COPS (perhaps nobody tried or they just gave up), but it is a known problem with "calibre-web". The autor of "calibre-web" did NOT officially solve it. It seems to be a problem with the (bugginess) of the tolino browser.

The workaroud seems to be (not tested/do not have a tolino):
`One way to work around the problem is to click the books author. Then from the author page click the selected books title (not the cover) to got the books page. At this point you can download the file.`

Can you try this?

Logs in a "linux" system (and the Synology is a linux system) ate typically written to **/var/log**


PS: I'm just a fellow COPS enthousiast. I know how to install and make COPS work. I do not know how COPS is programmed.

**Comment by cslondon** [comment]
Hello,
Thank you for the info, will give it a try when back from trip.
On the Synology the web server have special logging areas separate from standard Linux, haven't drilled down by now. Will also check further on the web browser error which seems most specific to the issue. Not sure if the server side gets aware of the problem on the client side.

On August 12, 2022 4:17:29 PM GMT+02:00, marioscube ***@***.***> wrote:
***@***.***

**Comment by marioscube** [comment]
@cslondon 
Do you have the same issues with the Tolino app for iOS/Android?

If "yes" then I can test with my version of COPS.
If "no" I will not bother installing it.

Sometimes, in my experience, COPS (apache/PHP) throws an error if a book cannot be found or cannot be delivered completely.
I do not have a Synology with DSM 7. MIne is still on DSM 5 :-(

**Comment by cslondon** [comment]
@marioscube 
I did the tests back from trip: The problem downloading the books is independent from where I try to access the book, it always starts downloading but fails with errormessage, so going via author doesn't change this. Tolino APP: I didn't see an option to connect the tolino app to COPS. Downloading from normal webbroser on tablets and PC  isn't a problem. But: I found  the reason for my problems: all readers (paperwhite and tolino) are blocked in my router to access the internet, they are only allowed local access including the synology with cops. Main reason for the blocker were some automatic firmware updates on the readers which happened which I couldn't dismiss/reject. The tolino webrowser has apparently a problem with this, when I remove the internet blocker on the router (AVM Fritzbox  under internet filters) it works now. So the tolino browser apparently does something besides the local cops access when downloading a book from cops. Didn't find anything further from the logs.
I did a test on a tablet with blocking there same way internet access and downloading a book, this is no problem with internet access blocked.
So the issue is tolino/webbrowser specific together with internet blocking, nothing specific to cops. 
Thanks again for your feedback and help.
